include/common.h

```diff
@@ -44,7 +50,7 @@ namespace ace {
         template<typename T>
         std::enable_if_t<std::is_enum<T>::value, T> choice_range(T first, T last) {
             using underlying = std::underlying_type_t<T>;
-            using type = std::conditional_t<std::is_same_v<signed char, std::make_signed_t<underlying>>, int, underlying>;
+            using type = std::conditional_t<std::is_same_v<signed char, std::make_signed_t<underlying>>::value, int, underlying>;
```

Comment:
As of C++17, `std::is_same_v<T, U>` is supposed to be an alias for `std::is_same<T, U>::value`. I believe this is my oversight, I thought this was a C++14 feature as a lot of other SFINAE aliases were (e.g. `std::underlying_type_t`, `std::conditional_t`, etc). I blame MSVC for allowing this in C++14 >:(

I think I'd like to keep this repo C++14 compliant for now, so it would be better to get rid of the macro and replace `std::is_same_v` with `std::is_same<T, U>::value`